lib/agent.js

```diff
@@ -128,11 +144,41 @@ function Agent (plugin) {
         return this.plugin.formatLine(metric)
       }
     },
+    generateGlobalTags: function (tagNames, tagObject) {
+      if (tagNames) {
+        if (typeof tagNames === 'string') {
+          // convert comma separated String list to array
+          tagNames = tagNames.replace(/\s/g, '').split(',')
```

Comment:
Can you add a regex check to make sure the structure of the tags is correct, and gracefully handle errors when the structure is not the way we predict it: `key: value, user, whatever: you-want` etc.

Reply:
If there is no `:` the value is interpreted as the name of an `env` variable. The replace regex is just to remove user-friendly white space.